### PR TITLE
Project board automation

### PR DESCRIPTION
**Pull request by evertonlperes**
### Changes
1. Implemented a GHA workflow to auto assign Project  in `opened/reopened`  issues

#### How it works?
The workflow will detect any new or reopened issues and will trigger the GHA runner. We don't need to create or edit the existing issues templates, the automation will use the existing ones.

Screen Shot 2022-01-28 at 1 52 22 PM
.

### CODE REVIEW

File: .github/workflows/board-automation.yaml
Comment (mook-as) [inline .github/workflows/board-automation.yaml:12]: For actions not from `actions/` (i.e. unofficial), we should pin to a revision so they can't change it and expose secrets.

File: .github/workflows/board-automation.yaml
Comment (mook-as) [inline .github/workflows/board-automation.yaml:17]: nit: new line at end of file.

**Review by mook-as — CHANGES_REQUESTED** [review]

File: .github/workflows/board-automation.yaml
Comment (evertonlperes) [reply .github/workflows/board-automation.yaml:12]: Sweet!
I didn't know about it. Thanks for sharing @mook-as 
I'll fix that.

File: .github/workflows/board-automation.yaml
Comment (evertonlperes) [reply .github/workflows/board-automation.yaml:17]: ough.. sorry, yaml linter was sleeping on that...

File: .github/workflows/board-automation.yaml
Comment (mook-as) [inline .github/workflows/board-automation.yaml:14]:
```suggestion
          github_token: ${{ github.token }}
```
We shouldn't need to create a new token for use with the action, when the action runs with its own temporary token anyway.

**Review by mook-as — CHANGES_REQUESTED** [review]
Please set workflow-level [permissions](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions) so that we know exactly which permissions the workflow has (and so that it's reduced to a minimal set).

I suspect it needs `issues: write` (to update the issue's associated projects), and possibly `repository-projects: read`.

File: .github/workflows/board-automation.yaml
Comment (evertonlperes) [reply .github/workflows/board-automation.yaml:14]: Fixed!

**Comment by evertonlperes** [comment]
> Please set workflow-level [permissions](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions) so that we know exactly which permissions the workflow has (and so that it's reduced to a minimal set).
> 
> I suspect it needs `issues: write` (to update the issue's associated projects), and possibly `repository-projects: read`.

Thanks for sharing. One more security level doesn't hurt :) 

Regarding the `repository-projects` tag, we need to set it as `write`, otherwise the automation cannot send a POST to move the issue to the `ToDo` column.

```json
Adding issue 13 to project 1

{
  data: { updateIssue: null },
  errors: [
    {
      type: 'FORBIDDEN',
      path: [Array],
      locations: [Array],
      message: "You don't have permission to add to project with id 'PRO_kwLODz3aws4A1qIF'."
    }
  ]
}
```

File: .github/workflows/board-automation.yaml
Comment (mook-as) [inline .github/workflows/board-automation.yaml:9]: Hmm, if we need `repository-projects: write` to move issues into a project, does that mean we only need `issues: read`? (Since it doesn't _quite_ make sense to need write on both sides… not sure, though.)

File: .github/workflows/board-automation.yaml
Comment (mook-as) [reply .github/workflows/board-automation.yaml:17]: It doesn't look like the fix was pushed, though…

**Review by mook-as — CHANGES_REQUESTED** [review]

File: .github/workflows/board-automation.yaml
Comment (evertonlperes) [reply .github/workflows/board-automation.yaml:17]: thanks for catching...
setting up vscode auto-save :P

File: .github/workflows/board-automation.yaml
Comment (evertonlperes) [reply .github/workflows/board-automation.yaml:9]: let's keep with the issues:write only... it makes more sense though.

Comment (mook-as) [reply .github/workflows/board-automation.yaml:9]: Can we get away with
```yaml
permissions:
    issues: read
    repository-projects: write
```

(The answer may be _no_; but it's useful to find out.)

**Review by mook-as — APPROVED** [review]
I'd like to see if we can reduce the permissions; approving anyway in case we can't (in which case we wouldn't need a re-approval).

**Comment by evertonlperes** [comment]
> I'd like to see if we can reduce the permissions; approving anyway in case we can't (in which case we wouldn't need a re-approval).

Gotcha.
Unfortunately we can't, we need to give the write access :( 

```bash
# issues: read
 Adding issue 15 to project 1

{
  data: { updateIssue: null },
  errors: [
    {
      type: 'FORBIDDEN',
      path: [Array],
      extensions: [Object],
      locations: [Array],
      message: 'Resource not accessible by integration'
    }
  ]
}
```

**Pull request merged**
